quadriga/src/main/java/edu/asu/diging/quadriga/aspect/AuthAspect.java

```
@@ -0,0 +1,54 @@
+package edu.asu.diging.quadriga.aspect;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+
+import edu.asu.diging.quadriga.config.web.TokenInfo;
+
+@Aspect
+@Configuration
+public class AuthAspect {
+
+    @Pointcut("execution(@edu.asu.diging.quadriga.aspect.annotation.InjectToken * *(..))")
+    public void annotatedMethod() {}
+
+    @Pointcut("execution(* *(.., edu.asu.diging.quadriga.config.web.TokenInfo, ..))")
+    public void tokenParamMethod() {}
+
+    @Around("annotatedMethod() && tokenParamMethod()")
```

since those two go hand in hand (annotating the method only makes sense if there is a token parameter), I would raise a runtime exception if there is no token param.